sanic/app.py

```diff
@@ -311,19 +316,37 @@ def register_middleware(
             **response** - Invoke before the response is returned back
         :return: decorated method
         """
-        if attach_to == "request":
+        retval = middleware
```

Comment:
Slightly tangential to this PR, but what's the point of returning the exact same object that we're receiving (even if we end up changing it/creating a new one)?

Reply:
TBH, there is no reason IMO to return anything. But we do. If we are, it certainly should not be modified (which we need to do with `priority` ordering) since that would lead to some very hard to discover bugs.